content/dotnet-template-azure-iot-edge-module/CSharp/ModuleBackgroundService.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Client.Transport.Mqtt;
+using System.Text;
+
+namespace SampleModule;
+
+internal class ModuleBackgroundService : BackgroundService
+{
+    private int counter;
+    private readonly ILogger<ModuleBackgroundService> _logger;
+
+    public ModuleBackgroundService(ILogger<ModuleBackgroundService> logger) => _logger = logger;
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        MqttTransportSettings mqttSetting = new(TransportType.Mqtt_Tcp_Only);
+        ITransportSettings[] settings = { mqttSetting };
+
+        // Open a connection to the Edge runtime
+        ModuleClient ioTHubModuleClient = await ModuleClient.CreateFromEnvironmentAsync(settings);
+        await ioTHubModuleClient.OpenAsync(stoppingToken);
+        _logger.LogInformation("IoT Hub module client initialized.");
+
+        // Register callback to be called when a message is received by the module
+        await ioTHubModuleClient.SetInputMessageHandlerAsync("input1", PipeMessage, ioTHubModuleClient, stoppingToken);
+    }
+
+    async Task<MessageResponse> PipeMessage(Message message, object userContext)
```

Comment:
No real reason to save the module client in the userContext because it can just as easily be a class field. It will make the code more readable because someone doesn't have to understand that it is being passed as an object and must be cast.